File: make_admin.py
import sys
from server import app, db, User

def make_admin(username):
    with app.app_context():
        user = User.query.filter_by(username=username).first()
        if not user:
            print(f"Error: User '{username}' not found.")
            return
        
        if user.is_admin:
            print(f"User '{username}' is already an admin.")
            return
            
        user.is_admin = True
        db.session.commit()
        print(f"Success: User '{username}' is now an admin.")

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python make_admin.py <username>")
        sys.exit(1)
    
    username = sys.argv[1]
    make_admin(username)

File: backend/scripts/make-admin.ts
/**
 * Script to promote a user to admin
 * Usage: npx tsx scripts/make-admin.ts <username>
 */

import { db, schema } from '../src/db/index.js';
import { eq } from 'drizzle-orm';

async function makeAdmin(username: string) {
  if (!username) {
    console.error('❌ Usage: npx tsx scripts/make-admin.ts <username>');
    process.exit(1);
  }

  // Find user
  const user = await db.query.users.findFirst({
    where: eq(schema.users.username, username),
  });

  if (!user) {
    console.error(`❌ User "${username}" not found`);
    process.exit(1);
  }

  if (user.isAdmin) {
    console.log(`✅ User "${username}" is already an admin`);
    process.exit(0);
  }

  // Update to admin
  await db.update(schema.users)
    .set({ isAdmin: true })
    .where(eq(schema.users.id, user.id));

  console.log(`✅ User "${username}" is now an admin!`);
  process.exit(0);
}

const username = process.argv[2];
makeAdmin(username);
